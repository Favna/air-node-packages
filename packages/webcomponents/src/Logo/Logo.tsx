import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import { customCss, dataQa } from '../constants';
import css from './Logo.scss';
import { ReactComponent as LogoSVG } from './logo.svg';
import React, { FC, Fragment, SVGProps, memo } from 'react';

export interface LogoProps extends SVGProps<any> {
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: dataQa;
  /** Custom CSS classes to pass to the button */
  customclasses?: customCss;
}

/** Creates a logo using pre-defined Rijkswatestaat SVG */
export const Logo: FC<LogoProps> = props => {
  return (
    <Fragment>
      <LogoSVG {...props} className={classnames(css.logo, css.customclasses)} />
    </Fragment>
  );
};

export default memo(Logo);