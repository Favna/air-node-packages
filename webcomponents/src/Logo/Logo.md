## Usage of the Logo component:

```tsx
import React, { FC } from 'react'

import { Logo } from '@rws-air/webcomponents'

const Example: FC = () => (
  <Logo />
)
```

**Regular size logo:**

```jsx
<Logo height={80}/>
```

**Very tiny logo:**

```jsx
<Logo height={10}/>
```