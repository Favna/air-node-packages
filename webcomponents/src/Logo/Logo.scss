.logo {
    height: 5.5rem !important;
    transform: translateX(50%) translateX(-23px) !important;
  }
  