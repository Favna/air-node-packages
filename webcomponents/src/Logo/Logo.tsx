import classnames from 'classnames';
import css from './Logo.scss';
import { ReactComponent as LogoSVG } from './logo.svg';
import React, { FC, Fragment, SVGProps } from 'react';

const Logo: FC<SVGProps<any>> = props => {
  return (
    <Fragment>
      <LogoSVG {...props} className={classnames(css.logo)} />
    </Fragment>
  );
};

export default Logo;