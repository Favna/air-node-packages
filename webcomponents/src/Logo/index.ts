export { default as Logo, LogoProps } from './Logo';