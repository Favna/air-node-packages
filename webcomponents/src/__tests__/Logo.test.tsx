import { shallow, ShallowWrapper } from 'enzyme';
import Logo from '../Logo/Logo';
import React, { SVGProps } from 'react';

let logo: ShallowWrapper<SVGProps<any>>;

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const logo = shallow(<Logo />);
    expect(logo).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const logo = shallow(<Logo height={500} width={500} cursor='pointer' />);
    expect(logo).toMatchSnapshot();
  });
});