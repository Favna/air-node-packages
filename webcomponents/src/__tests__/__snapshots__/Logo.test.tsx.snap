// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Optional Props 1`] = `
<Fragment>
  <ForwardRef(SvgLogo)
    className=""
    cursor="pointer"
    height={500}
    width={500}
  />
</Fragment>
`;

exports[`Snapshot Testing Required Props 1`] = `
<Fragment>
  <ForwardRef(SvgLogo)
    className=""
  />
</Fragment>
`;
